### PR TITLE
feat(grpc): --production

### PR DESCRIPTION
**Pull request by aarnphm**
this PR also address adding generic handlers from users if they want to add it
```
svc.add_generic_handlers(MyGrpcHandler(...))
```

address prometheus in a following PR.

### CODE REVIEW

File: bentoml/_internal/cli/bento_server.py [hunk 1]
```diff
@@ -53,6 +53,13 @@ def add_serve_command(cli: click.Group) -> None:
         help="The maximum number of pending connections.",
         show_default=True,
     )
+    @click.option(
+        "--max-concurrent-streams",
+        type=click.INT,
+        default=BentoMLContainer.grpc.max_concurrent_streams.get(),
+        help="Maximum number of concurrent incoming streams to allow on a HTTP/2 connection.",
+        show_default=True,
+    )
```
Comment (aarnphm) [inline bentoml/_internal/cli/bento_server.py:62]: TODO: implement this.

For now put it as stub.

File: bentoml/_internal/configuration/containers.py [hunk 2]
```diff
@@ -110,6 +110,7 @@ def _is_ip_address(addr: str) -> bool:
                 "access_control_expose_headers": Or([str], str, None),
             },
             "grpc": {
+                "max_concurrent_streams": Or(int, None),
```
Comment (aarnphm) [inline bentoml/_internal/configuration/containers.py:113]: This is the equivalent of `backlog` for uvicorn.

File: bentoml/_internal/server/grpc_app.py [hunk 3]
```diff
@@ -146,7 +173,7 @@ def interceptors(self) -> list[aio.ServerInterceptor]:
             AsyncOpenTelemetryServerInterceptor as AsyncOtelInterceptor,
         )
 
-        interceptors: list[t.Type[aio.ServerInterceptor]] = [
+        interceptors: list[t.Callable[[], aio.ServerInterceptor]] = [
```
Comment (aarnphm) [inline bentoml/_internal/server/grpc_app.py:176]: types for interceptor.

**Review by aarnphm — COMMENTED** [review]
TODO:
- refactor for grpc config
- interceptors, handlers shouldn't live under grpcapp.

**Review by aarnphm — COMMENTED** [review]
TODO:
- refactor for grpc config
- interceptors, handlers shouldn't live under grpcapp.

File: bentoml/_internal/server/cli/grpc_api_server.py [hunk 4]
```diff
@@ -0,0 +1,117 @@
...
+        params["worker_id"] = "$(circus.wid)"
+        watcher = Watcher(
+            name="bento_api_server",
+            cmd=sys.executable,
+            args=["-m", "bentoml._internal.server.cli.grpc_api_server"]
+            + unparse_click_params(params, ctx.command.params, factory=str),
+            copy_env=True,
+            numprocesses=1,
+            stop_children=True,
+            use_sockets=True,
```
Comment (bojiang) [inline bentoml/_internal/server/cli/grpc_api_server.py:89]:
```suggestion
```

**Review by bojiang — APPROVED** [review]
the code style failure is not introduced by this pr

**Pull request merged**